bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/firmware/FirmwareImpl.java

```diff
@@ -72,18 +75,20 @@
      * @param description the description of the firmware (can be null)
      * @param version the version of the firmware (not null)
      * @param prerequisiteVersion the prerequisite version of the firmware (can be null)
+     * @param firmwareRestriction {@link FirmwareRestriction} for applying an additional restriction on
+     *            the firmware (can be null). If null, a default function will be used to return always true
```

Comment:
IMHO The part `a default function will be used to return always true` is an implementation detail. It is enough to state that if the parameter is null, no additional restriction is applied.